### ISSUE TITLE
Pre-submit testing and CI for CSI

### ISSUE THREAD

**Issue by saad-ali** [container-storage-interface/spec#501]
CSI was previously using travis-ci for PR pre-submit testing and CI for this project. However travis-ci.org has been turned down and https://travis-ci.org/github/container-storage-interface/spec shows:

```
Since June 15th, 2021, the building on [travis-ci.org](http://www.travis-ci.org/) is ceased. Please use [travis-ci.com](http://www.travis-ci.com/) from now on.
```

We need a replacement pre-submit and CI solution. Opening this bug to track that work.

**Comment by saad-ali** [comment]
@pohly would you be interested in helping with this?

CC @xing-yang

**Comment by pohly** [comment]
Sorry, kind of busy right now...

**Comment by saad-ali** [comment]
> Sorry, kind of busy right now...

No worries, thanks for the reply @pohly

**Comment by jdef** [comment]
FWIW it might be nice to leverage github actions/workflows for this

**Comment by saad-ali** [comment]
CC @mattcary

**Comment by andyzhangx** [comment]
fixed by https://github.com/container-storage-interface/spec/pull/502

**Comment by mattcary** [comment]
Cool, thanks Andy!

**Comment by jdef** [comment]
The build badge in the README still links to travis. Before closing out this ticket, we should probably update the badge.

https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

**Comment by saad-ali** [comment]
Closed by #502 and #505 
Thanks Andy!
